pandas/tests/indexes/test_base.py

```diff
@@ -1604,3 +1604,15 @@ def test_get_attributes_dict_deprecated():
     with tm.assert_produces_warning(DeprecationWarning):
         attrs = idx._get_attributes_dict()
     assert attrs == {"name": None}
+
+
+def test_nan_comparison_same_object():
```

Comment:
can you test gt as well as lt; this doesn't affect le, ge right?

Reply:
Done and no, they are checked i a different if block